### PR TITLE
[News card] UI and integration in Reader

### PR DESCRIPTION
**Pull request by ctarda**
Implements #9908 and #9909 

simulator screen shot - iphone 8 - 2018-08-08 at 13 35 03
simulator screen shot - iphone 8 - 2018-08-08 at 13 38 27

In this PR:
- Implementation of the UI for the News Card
- Integration of the news card in the Reader headers. 
- The News Card should support RTL languages
- The News Card should be accessible.

This PR does not include any logic to decide when the card is visible, or when it can dismissed:
- The dismiss button does not work yet
- The "Read More" button does not work yet 
- The News Card would be always visible if the feature flag is active.

Implementing that logic would be done in another PR.

To test:
1. Without activating the feature flag, the news card should not be visible.
2. Activate the feature flag.
3. The news card should be visible in the Discover stream
4. The header of the Discover stream should also include the previous header

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/NewsCard.swift [hunk 1]
```diff
@@ -15,6 +24,124 @@ final class NewsCard: UIViewController {
...
+extension NewsCard: Accessible {
+    func prepareForVoiceOver() {
+        prepareDismissButtonForVoiceOver()
+        prepareReadMoreButtonForVoiceOver()
+    }
+
+    private func prepareDismissButtonForVoiceOver() {
+        dismiss.accessibilityLabel = NSLocalizedString("Dismiss", comment: "Accessibility label for the Dismiss button on Reader's News Card")
+        dismiss.accessibilityTraits = UIAccessibilityTraitButton
+        dismiss.accessibilityHint = NSLocalizedString("Dimisses the News Card.", comment: "Accessibility hint for the dismiss button on Reader's News Card")
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/NewsCard.swift]: Typo – `Dimisses` > `Dismisses`

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 2]
```diff
@@ -497,27 +497,30 @@ import WordPressFlux
...
             tableView.tableHeaderView = nil
             return
         }
 
-        header.enableLoggedInFeatures(isLoggedIn)
-        header.configureHeader(topic)
-        header.delegate = self
+        tableView.tableHeaderView = header
+
+        // This feels somewhat hacky, but it is the only way I found to insert a stack view into the header whitput breaking the autolayout constraints.
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Nitpick: `whitput` -> `without`

**Comment by ctarda** [comment]
Thanks for the review @frosty !

I've tackled most of your comments, however, I have not been able to figure out (yet) why that part of the header does not comply to the readable content guides, and I am getting a bit (actually a lot) frustrated. Considering this still requires a design review, and that it is not visible to users now because it is behind a feature flag, do you think we can move on for now?

One more thing, the news card reader should show (for now) at the top of every stream. Sorry, the PR description was confusing. The next step would be deal with the logic to decide when it is visible.

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 3]
```diff
@@ -504,7 +504,7 @@ import WordPressFlux
 
         tableView.tableHeaderView = header
 
-        // This feels somewhat hacky, but it is the only way I found to insert a stack view into the header whitput breaking the autolayout constraints.
+        // This feels somewhat hacky, but it is the only way I found to insert a stack view into the header whitout breaking the autolayout constraints.
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Almost! 😅 
`whitout` -> `without`

**Comment by ctarda** [comment]
😊 Sorry for that typo. Pushed an update and will merge now!

**Pull request merged**